lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -232,3 +233,6 @@ instance CompareDiscovery (RndState n) where
 
 instance KnownAddresses (RndState n) where
     knownAddresses s = Map.elems (addresses s)
+
+instance HasRewardAccount (RndState n) where
+    rewardAccount _ = Nothing
```

Comment:
I'm not sure about the `Has`-prefix. Seeing the constraint `HasRewardAccount s` I would like to believe that `s` **has** a reward account. But it doesn't mean that! `rewardAccount @s` could return `Nothing`!

Reply:
Why could it though ? I don't think `HasRewardAccount` should be defined on `RndState` in the first place actually.

Reply:
That's kinda the point of all these instances and the way the wallet engine works. You only define the capabilities on the relevant concrete types. 

`RndState` doesn't implement `DelegationAddress` for instance, and that's very fine :)

Reply:
> I don't think HasRewardAccount should be defined on RndState in the first place actually.

No, I agree. 

It simplified things in `Api.Server`. I will see now if I can do anything.

Reply:
Changed